### PR TITLE
Plan-dumper: Add Umbra, CedarDB and MariaDB

### PR DESCRIPTION
**Pull request by vogelsgesang**
I am planning to add support for additional database engines. In preparation for that, add reproducible query-plan fixtures for Umbra, CedarDB, and MariaDB.

- Add plain and analyzed plans for all three engines.
- Add CedarDB optimizer-phase snapshots and MariaDB optimizer traces.
- DuckDB plans were also re-generated. They contain more details now.
- Use engine-specific SQL setup scripts for schema creation, data loading, and statistics collection.
- Add `\include` and `\copy` setup directives where native SQL loading is insufficient.
- Improve filtering of volatile runtime measurements for new databases.
- Remove obsolete manually maintained Hyper fixtures.

### CODE REVIEW

File: plan-dumper/queries/tpch/tpch-q15.sql [hunk 1]
```diff
@@ -1,3 +1,4 @@
+-- UNSUPPORTED: cedardb
```
Comment (vogelsgesang) [inline plan-dumper/queries/tpch/tpch-q15.sql:1]: why? Add comment

File: plan-dumper/queries/cte-recursive.sql [hunk 2]
```diff
@@ -1,3 +1,6 @@
+-- UNSUPPORTED: trino:analyze
+-- Trino's default recursion limit rejects this 100-step query; raising the limit
+-- instead causes a stack overflow during analysis on Trino 434.
```
Comment (vogelsgesang) [inline plan-dumper/queries/cte-recursive.sql]: change this query to use a smaller iteration count, so we can also test it on Trino

File: plan-dumper/queries/insert.sql [hunk 3]
```diff
@@ -1,3 +1,4 @@
+-- UNSUPPORTED: mariadb
```
Comment (vogelsgesang) [inline plan-dumper/queries/insert.sql:1]: add comment why it's unsupported

File: plan-dumper/queries/metadata-describe-table.sql [hunk 4]
```diff
@@ -1,4 +1,4 @@
--- UNSUPPORTED: duckdb
+-- UNSUPPORTED: duckdb, mariadb, trino, umbra
 -- MODES: simple
 -- Uses Postgres catalog casts (`::pg_catalog.regtype::pg_catalog.text`) that DuckDB's
 -- pg_catalog emulation doesn't support; Hyper's does, since it's Postgres-wire-compatible.
```
Comment (vogelsgesang) [inline plan-dumper/queries/metadata-describe-table.sql]: update comment to cover newly unsupported databases

File: plan-dumper/queries/tablefunction.sql [hunk 5]
```diff
@@ -1,2 +1,3 @@
+-- UNSUPPORTED: mariadb, trino
```
Comment (vogelsgesang) [inline plan-dumper/queries/tablefunction.sql]: should we switch to some other function which might work on all databases?

File: plan-dumper/queries/tpch-q11-error.sql
Comment (vogelsgesang) [inline plan-dumper/queries/tpch-q11-error.sql]: udpate comment to also cover other databases

File: plan-dumper/filter-timing-diff.py [hunk 6]
```diff
@@ -22,7 +22,10 @@
 import sys
 
 VOLATILE_FIELD_RE = re.compile(
-    r'"[^"]*(?:time|timing|latency|cycles)[^"]*"\s*:\s*[-+0-9.eE]+', re.IGNORECASE
+    r'"(?:[^"]*(?:time|timing|latency|cycles|duration)[^"]*'
+    r'|(?:addInput|getOutput|finish|blocked)(?:Cpu|Wall)|start|stop)"'
+    r'\s*:\s*(?:"(?:\\.|[^"\\])*"|[-+0-9.eE]+)',
+    re.IGNORECASE,
```
Comment (vogelsgesang) [inline plan-dumper/filter-timing-diff.py]: add comment clarifying which exact keys each database uses

File: plan-dumper/requirements.txt [hunk 7]
```diff
@@ -1,3 +1,5 @@
 tableauhyperapi
-psycopg2
+psycopg2-binary
```
Comment (vogelsgesang) [inline plan-dumper/requirements.txt:2]: why switch to binary?

File: plan-dumper/README.md [hunk 8]
```diff
@@ -19,45 +19,96 @@ A `-- MODES: <mode>[, <mode>...]` comment in a query file lists the `EXPLAIN` mo
...
-* `analyze` → `EXPLAIN (…, ANALYZE)` — includes runtime measurements (Hyper, Postgres and DuckDB).
-* `steps` → the optimizer's intermediate steps: `EXPLAIN (FORMAT INTERNAL, OPTIMIZE STEPS)` on Hyper, `EXPLAIN (FORMAT JSON)` with `explain_output='all'` on DuckDB (skipped for Postgres).
-* `pipelines` / `analyze-pipelines` → execution pipelines (Hyper only; skipped for Postgres and DuckDB).
+* `simple` → a plain JSON `EXPLAIN`.
+* `analyze` → an executing plan with runtime measurements. Hyper, Postgres, DuckDB, Umbra, CedarDB, and MariaDB use their JSON `EXPLAIN ANALYZE` variants. For Trino, the dumper executes the query and fetches its raw, unpruned `QueryInfo` JSON from the coordinator; this intentionally has a different format from Trino's simple JSON plan.
+* `steps` → optimizer internals: `EXPLAIN (FORMAT INTERNAL, OPTIMIZE STEPS)` on Hyper; logical/optimized/physical plans on DuckDB; Umbra's ordered `debug.optimizer.steplog` events; all ten native `EXPLAIN (STEP ..., FORMAT JSON)` plan snapshots on CedarDB; and MariaDB's JSON `optimizer_trace`. MariaDB's output is a decision trace rather than a sequence of complete plans. Postgres and Trino currently skip this mode.
```
Comment (vogelsgesang) [inline plan-dumper/README.md]: elaborate in more detail; also, clarify DuckDB further in the list than I did

```suggestion
* `steps` → optimizer internals: 
  * Hyper: EXPLAIN (FORMAT INTERNAL, OPTIMIZE STEPS)`
  * DuckDB: logical/optimized/physical plans
  * Umbra's: `SET debug.optimizer.steplog` scrape event from the log events
  * CedarDB: `EXPLAIN (STEP ..., FORMAT JSON)`
  * MariaDB's JSON `optimizer_trace`. MariaDB's output is a decision trace rather than a sequence of complete plans.
  * Postgres and Trino currently skip this mode.
```

File: plan-dumper/README.md [hunk 9]
```diff
@@ -19,45 +19,96 @@ A `-- MODES: <mode>[, <mode>...]` comment in a query file lists the `EXPLAIN` mo
...
+* `pipelines` / `analyze-pipelines` → execution pipelines (Hyper only).
 
 Each requested mode is written to its own `<query>[-<mode>].plan.json` (no suffix for `simple`).
 A mode not supported by a given database is silently skipped and no file is generated.
 
-If a database can't run a query at all, mark it with a `-- UNSUPPORTED: <db>[, <db>...]` comment instead.
+If a database can't run a query at all, mark it with a `-- UNSUPPORTED: <db>[, <db>...]` comment. Use `<db>:<mode>` when only one mode is unsupported. Any unannotated query failure makes the run fail and leaves that database's previous output directory untouched.
```
Comment (vogelsgesang) [inline plan-dumper/README.md]: Split across multiple lines

```suggestion
If a database can't run a query at all, mark it with a `-- UNSUPPORTED: <db>[, <db>...]` comment.
Use `<db>:<mode>` when only one mode is unsupported.
Any unannotated query failure makes the run fail and leaves that database's previous output directory untouched.
```

File: plan-dumper/README.md [hunk 10]
```diff
@@ -19,45 +19,96 @@ A `-- MODES: <mode>[, <mode>...]` comment in a query file lists the `EXPLAIN` mo
...
 
-**Prerequisites:**
+Install the Python dependencies:
 
 ```shell
 pip3 install -r plan-dumper/requirements.txt
 ```
 
-**With the default (pip-installed) Hyper:**
+Hyper and DuckDB are embedded. Postgres defaults to port 5432; the other server databases are enabled through command-line options. With every server configured, one invocation dumps all databases:
```
Comment (vogelsgesang) [inline plan-dumper/README.md]: Here and other places: one sentence per line

```suggestion
Hyper and DuckDB are embedded. Postgres defaults to port 5432; the other server databases are enabled through command-line options.
With every server configured, one invocation dumps all databases:
```

File: plan-dumper/README.md [hunk 11]
```diff
@@ -19,45 +19,96 @@ A `-- MODES: <mode>[, <mode>...]` comment in a query file lists the `EXPLAIN` mo
...
+python3 dump-plans.py \
+  --postgres-dsn "host=127.0.0.1 port=5432 user=postgres" \
+  --umbra-dsn "host=127.0.0.1 port=5433 user=postgres password=postgres" \
+  --cedardb-dsn "host=127.0.0.1 port=5435 user=postgres password=CedarDB1!" \
+  --mariadb-url "mysql://plan_dumper:password@127.0.0.1/query_graphs_plan_dumper" \
+  --trino-url "http://plan-dumper@127.0.0.1:8080/memory/query_graphs_plan_dumper"
 ```
 
-**With a custom Hyper build:**
+Omit an option to skip Umbra, CedarDB, MariaDB, or Trino. Pass `--postgres-dsn ""` to skip Postgres. A connection failure also skips that database without preventing the remaining dumpers from running.
```
Comment (vogelsgesang) [inline plan-dumper/README.md]: why is Postgres different from the others? We probably want to align Postgres with the others

File: plan-dumper/filter-timing-diff.py [hunk 12]
```diff
@@ -1,11 +1,10 @@
...
 
-Regenerating the example plans re-runs every query, so timing/cycle-count fields (e.g.
-Postgres' "Actual Total Time", DuckDB's "cpu_time", Hyper's "cpu-cycles") change on every
-run even when the plan itself didn't. A hunk is dropped if every removed line matches its
-added counterpart once those volatile fields are masked out; if a line differs for any
-other reason, the whole line (including its timing/cycle-count change) is kept as-is.
+Regenerating the example plans re-runs every query, so timing, memory, and scheduler
+fields change on every run even when the plan itself didn't. A hunk is dropped if every
+removed line matches its added counterpart once those volatile fields are masked out;
+if a line differs for any other reason, the whole line is kept as-is.
```
Comment (vogelsgesang) [inline plan-dumper/filter-timing-diff.py]:
```suggestion
fields change on every run even when the plan itself didn't. Hunk which only differ on
those volatile fields are dropped. If a hunk differs for any other reason, the whole line is kept as-is.
```

File: plan-dumper/filter-timing-diff.py [hunk 13]
```diff
@@ -21,23 +20,58 @@
...
 import sys
 
-VOLATILE_FIELD_RE = re.compile(
-    r'"[^"]*(?:time|timing|latency|cycles)[^"]*"\s*:\s*[-+0-9.eE]+', re.IGNORECASE
+JSON_SCALAR_RE = r'\s*:\s*(?:"(?:\\.|[^"\\])*"|[-+0-9.eE]+|true|false|null)'
+
+# Postgres uses keys such as "Actual Total Time"; Hyper uses "cpu-cycles";
+# DuckDB uses "cpu_time", "operator_timing", and "latency"; MariaDB uses
+# "r_total_time_ms"/"r_table_time_ms"; Umbra uses "durationUs"; and Trino uses
+# "*Time", "*Cpu", and "*Wall" keys.
```
Comment (vogelsgesang) [inline plan-dumper/filter-timing-diff.py]: use bullet point list:


```suggestion
# Volatile fields:
# * Hyper: `cpu-cycles`;
# * Umbra uses `durationUs`;
# * Postgres: `Actual Total Time`
# * DuckDB: `cpu_time`, `operator_timing`, and `latency`
# * Trino: `*Time`, `*Cpu`, and `*Wall` keys
# * MariaDB: `r_total_time_ms`/`r_table_time_ms`
```

File: plan-dumper/dump-plans.py [hunk 14]
```diff
@@ -1,171 +1,685 @@
...
-}
+try:
+    import pymysql
+except ImportError:
+    pymysql = None
 
+try:
+    import trino
+except ImportError:
+    trino = None
```
Comment (vogelsgesang) [inline plan-dumper/dump-plans.py]: no need to try-import; just assume that the `requirements.txt` got installed

File: plan-dumper/dump-plans.py [hunk 15]
```diff
@@ -1,171 +1,677 @@
...
-parser = argparse.ArgumentParser()
-parser.add_argument("--hyper-path", type=Path, default=None,
-                    help="Path to a directory containing the hyperd binary. "
-                         "Uses the pip-installed Hyper by default.")
-args = parser.parse_args()
 
+baseDir = Path(__file__).resolve().parent
+setupFile = baseDir / "setup.sql"
+queriesDir = baseDir / "queries"
+targetDir = baseDir.parent / "standalone-app" / "examples"
```
Comment (vogelsgesang) [inline plan-dumper/dump-plans.py]: here and rest of file: align naming with Python best practices
